src/earn/EarnEnterAmount.tsx

```diff
@@ -81,17 +81,39 @@ function useTokens({ pool, mode }: { pool: EarnPosition; mode: EarnDepositMode }
     throw new Error(`Token info not found for token ID ${pool.dataProps.depositTokenId}`)
   }
 
-  return mode === 'deposit' ? [depositToken] : eligibleSwappableTokens
+  if (!withdrawToken) {
+    // should never happen
+    throw new Error(`Token info not found for token ID ${pool.dataProps.withdrawTokenId}`)
+  }
+
+  return {
+    depositTokens: [depositToken],
+    withdrawTokens: [withdrawToken],
```

Comment:
my main intent was to avoid having something that could mean there can be multiple `deposit` / `withdraw` tokens, which isn't the case now. There will never be multiple withdraw tokens, maybe multiple deposit in the future, but for now the backend keeps it singular. And the reason `useTokens` was previously returning an array of tokens was because it was just the list that goes in the token selector, but now the purpose of this function has changed explicitly to return deposit / withdraw / swap tokens. So I'd favor changing deposit / withdraw to singular to avoid any ambiguity in the future.

Reply:
Fixed in b989f051f0d1c71a9527d9689b7f230648ed94f2!